lms/djangoapps/instructor/tests/views/test_instructor_dashboard.py

```diff
@@ -135,6 +140,33 @@ def has_instructor_tab(user, course):
         )
         assert has_instructor_tab(org_researcher, self.course)
 
+    @ddt.data(
+        ('staff', False, False, True),
+        ('staff', True, False, False),
+        ('staff', True, True, True),
+    )
+    @ddt.unpack
+    def test_discussion_tab(self, access_role, is_pages_and_resources_enabled, is_legacy_discussion_setting_enabled,
+                            is_discussion_tab_available):
+        """
+        Verify that the Discussion tab only shows up when Pages & Resources flag is off for course
+        """
+        discussion_section = '<li class="nav-item"><button type="button" class="btn-link discussions_management" ' \
+                             'data-section="discussions_management">Discussions</button></li>'
+
+        with override_waffle_flag(ENABLE_PAGES_AND_RESOURCES_MICROFRONTEND, is_pages_and_resources_enabled):
+            with override_waffle_flag(OVERRIDE_DISCUSSION_LEGACY_SETTINGS_FLAG, is_legacy_discussion_setting_enabled):
+                CourseAccessRoleFactory(
```

Comment:
we need two other cases here.
`test_discussion_tab_for_global_staff`
`test_discussion_tab_for_course_instructor`